pkg/domain/infra/abi/farm.go

```diff
@@ -27,15 +27,13 @@ func (ir *ImageEngine) FarmNodeDriver(ctx context.Context) string {
 
 func (ir *ImageEngine) fetchInfo(_ context.Context) (os, arch, variant string, nativePlatforms []string, emulatedPlatforms []string, err error) {
 	nativePlatform := parse.DefaultPlatform()
-	platform := strings.SplitN(nativePlatform, "/", 3)
-	switch len(platform) {
-	case 0, 1:
+	var hasArch bool
+	os, arch, hasArch = strings.Cut(nativePlatform, "/")
+	if !hasArch {
 		return "", "", "", nil, nil, fmt.Errorf("unparsable default platform %q", nativePlatform)
-	case 2:
-		os, arch = platform[0], platform[1]
-	case 3:
-		os, arch, variant = platform[0], platform[1], platform[2]
 	}
+	arch, variant, _ = strings.Cut(arch, "/")
```

Comment:
Just how bad is it to use `SplitN()`? Like, is it deprecated and I just can't see where? Because this here is more readable (to me, of course) in the original `SplitN() + switch` form.

Reply:
SplitN() is fine and not deprecated.
I think the main issue is with people using SplitN incorrectly (i.e. forget to check length and just access the array and then run into a panic due out of bound access). Overall I think this makes it much better but here I tend to agree using SplitN(...,3) here seems more straight forward for the reader to undersatnd.

Reply:
It isn't deprecated. Cut was introduced because it tends to be what one wants in the majority of cases when using SplitN. Cut avoids allocating a slice while making most code simpler

Reply:
Reverted changing this file